js/gateio.js

```diff
@@ -3125,6 +3127,230 @@ module.exports = class gateio extends Exchange {
+            Delivery Futures
+            {
+                "name": "BTC_USDT_20200814",
+                "underlying": "BTC_USDT",
+                "cycle": "WEEKLY",
+                "type": "direct",
+                "quanto_multiplier": "0.0001",
+                "mark_type": "index",
+                "last_price": "9017",
+                "mark_price": "9019",
+                "index_price": "9005.3",
+                "basis_rate": "0.185095",
+                "basis_value": "13.7",
+                "basis_impact_value": "100000",
+                "settle_price": "0",
+                "settle_price_interval": 60,
+                "settle_price_duration": 1800,
+                "settle_fee_rate": "0.0015",
+                "expire_time": 1593763200,
+                "order_price_round": "0.1",
+                "mark_price_round": "0.1",
+                "leverage_min": "1",
+                "leverage_max": "100",
+                "maintenance_rate": "1000000",
+                "risk_limit_base": "140.726652109199",
+                "risk_limit_step": "1000000",
+                "risk_limit_max": "8000000",
+                "maker_fee_rate": "-0.00025",
+                "taker_fee_rate": "0.00075",
+                "ref_discount_rate": "0",
+                "ref_rebate_rate": "0.2",
+                "order_price_deviate": "0.5",
+                "order_size_min": 1,
+                "order_size_max": 1000000,
+                "orders_limit": 50,
+                "orderbook_id": 63,
+                "trade_id": 26,
+                "trade_size": 435,
+                "position_size": 130,
+                "config_change_time": 1593158867,
+                "in_delisting": false
+            }
+            @param market: CCXT market
+        */
+        const maintenanceMarginUnit = this.safeString (info, 'maintenance_rate'); // '0.005',
+        const leverageMax = this.safeString (info, 'leverage_max'); // '100',
+        const riskLimitStep = this.safeString (info, 'risk_limit_step'); // '1000000',
+        const riskLimitMax = this.safeString (info, 'risk_limit_max'); // '16000000',
+        const initialMarginUnit = Precise.stringDiv ('1', leverageMax);
+        let maintenanceMarginRate = maintenanceMarginUnit;
+        let initialMarginRatio = initialMarginUnit;
+        let floor = '0';
+        const tiers = [];
+        while (Precise.stringLt (floor, riskLimitMax)) {
+            const cap = Precise.stringAdd (floor, riskLimitStep);
+            tiers.push ({
+                'tier': this.parseNumber (Precise.stringDiv (cap, riskLimitStep)),
+                'currency': this.safeString (market, 'settle'),
+                'notionalFloor': this.parseNumber (floor),
+                'notionalCap': this.parseNumber (cap),
+                'maintenanceMarginRate': this.parseNumber (maintenanceMarginRate),
+                'maxLeverage': this.parseNumber (Precise.stringDiv ('1', initialMarginRatio)),
+                'info': info,
```

Comment:
@Dan-krm On gateio for example, the leverage tiers are calculated from `risk_limit_step`, `risk_limit_max`, `leverage_max`, and `maintenance_rate`